### ISSUE TITLE
`readability-static-accessed-through-instance` not reported in templated function

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#129591]
```cpp
class C
{
public:
    enum E { E1 };
    static void f();

    E e;
};

template<class T>
void f(const T& c)
{
    if (c.e == c.E1) {}
    c.f();
}
```

https://godbolt.org/z/sPGqdjKM4

**Comment by localspook** [comment]
I believe the check is right to not diagnose this. If the function template is instantiated with a `T` whose `f` member function or `E1` variable isn't static, changing `c.f()` to `T::f()` or `c.E1` to `T::E1` would break that.
